### DISCUSSION TITLE
Gifs dont play when using ImageRequest target onSuccess

### DISCUSSION THREAD

**Discussion by tyczj** [coil-kt/coil#1080] [Q&A]
I am trying to load a gif from file like this

    _contentImage.load(File(image))
        {
            crossfade(true)
            listener(onError = { _, error ->
                Log.d(SUBTAG, "Error -> ${error.message}")
            },
                onSuccess = { _, _ ->
                    Log.d(SUBTAG, "Content image loaded")
                    contentImageLoaded = true
                    checkContentLoaded()
                })
            target(
                onSuccess = { result ->
                    Palette.from(result.toBitmap()).generate {
                        if (it != null) {
                            val paletteColor = it.getDominantColor(color)
                            _constraintLayout.setBackgroundColor(paletteColor)
                        }
                    }
                    _contentImage.setImageBitmap(result.toBitmap())
                }
            )
        }

This works fine for images but if the file is a gif the gif does not animate. If I remove the target section of the ImageRequest builder the gif plays as expected.

What I suspect is the issue here is the line `_contentImage.setImageBitmap(result.toBitmap())` in the onSuccess but this is the only way I was able to actually get images to show in the imageview, without that nothing ever appeared so perhaps I am doing something wrong there causing gifs not to animate?

**Comment by colinrtwhite** [comment]
You're converting the drawable to a bitmap which is a static image. You need to set the drawable on the `ImageView` and do `(result as Animatable).start()`.
